### DISCUSSION TITLE
GitHub is closed source, why still use it by referring "Open Source" while code will be hosted on a "Closed Source" cloud platform

### DISCUSSION THREAD

**Discussion by 0xfffffffffffffffffffffffe150** [community/community#19990] [Other Feature Feedback, Questions, & Ideas]
Good evening my dear Sisters and Brothers! As we know that GitHub is now owned by Microsoft and it's a proprietary (closed source) software or web platform. I wonder why using GitHub instead other platforms such as GitLab, now reasons might be like GitHub have more "features" than other ones, more engagement etc. But besides those what is the exact reason to use it?

Thank you!

**Comment by 0xfffffffffffffffffffffffe150** [comment]
I wonder there are no answer.

**Comment by diegoleondev** [comment]
### Main reason

Github is the platform that has been in operation for the longest time, and it has been managed to position, with the purchase of Microsoft increased the loyalty of the platform since now there is an important company behind, this means that It won't go away overnight, which gives companies security. 

Being a requirement in most companies, it is good to learn the Github environment.

### More details
- An extensive community
- Multiple integrations with third-party applications
- Security
- Error tracking.
- Quick search

# En resumen Github es el mejor 🌟

**Comment by ghost** [comment]
Just use gitlab ;)

**Reply by jeffmcneill** [reply]
Gitlab is closed source, unless you mean the self-hosted Gitlab.

**Reply by DitoopCX** [reply]
No, I won't ;)

**Comment by CRC-Mismatch** [comment]
If you're not happy with GitHub being a closed-source project providing free hosting for and access to open-source projects, feel free to avoid it :man_shrugging:... Either way, as @diegoleondev pointed out, it's become a _de facto_ standard market VCS provider, and most corporations have it as a standard for job-opening candidacy application (sometimes only using it as a "test-ground" while they themselves host their projects under other providers or in-house).

Although this may be contested, (probably due to the extensive popularity) GitHub has also achieved a _de facto_ standard status for CI/CD with GitHub Actions. I see it as "Arduino (although open-sourced - so whether the framework or the boards themselves) have become the 'standard' for embedded projects, Raspberry Pi has become a standard for full-fledged SBC/embedded applications, GitHub has become a standard for VCS" in a market-wise POV :man_shrugging:.

There are many other SBC or user-friendly-multi-purpose-proto-MCU-board projects, some with better aspects, some with cheaper aspects, as well as other (Git or otherwise) host services, but these've become the _de facto_ standard, and that's literally factual :man_shrugging:.

Also, GitLab may be open-source, but they also host closed-source projects, and thus I see it as a six-for-half-a-dozen situation :man_shrugging:

**Reply by jeffmcneill** [reply]
Well, "standards" don't just stick around forever, and so the use of Codeberg / Forgejo is better for those who care about open source.

**Comment by aborisovich** [comment]
Reason to use Github is the fact that is was bought by Microsoft, and as all "payed commercial solutions" provide better quality over "free open source" ones! 😁

But seriously, why not try to use it and make your own opinion? You know people are using it because they like it, that's how we humans are built.

**Reply by CRC-Mismatch** [reply]
Or because they "have to", due to external reasons (like work requirements and such). But still, since it's a big Corpo behind it, all other corporations that "look after themselves" prefer it just for that reason, since this (at lest seems to imply that it) creates a sense of security (or at the very least, accountability in a worst case scenario)...

**Reply by jeffmcneill** [reply]
There are two problems with this statement: 1. that Github's open/closed/free/paid status changed after Microsoft bought it (no, it did not it was basically the same before it was purchased and after it was purchased, and 2. that "payed" (I think you mean proprietary) solutions are higher quality than open source ones. The success of projects such as Linux (open source), Android, etc. put the lie to that idea. If you look at the latest shenanigans regarding Cloudstrike and Windows, can you really say there is better security, much less accountability?

**Reply by theist** [reply]
tbh cloudstrike pinned down Debian systems way before Windows ones, but nobody noticed  https://stackdiary.com/crowdstrike-took-down-debian-and-rocky-linux-a-few-months-ago-and-no-one-noticed/

**Comment by WalliGateshead** [comment]
Good evening! While it is true that Microsoft owns GitHub, it is not accurate to say that it is a closed-source platform. GitHub's core product is built on top of Git, an open-source version control system, and it allows users to host their code repositories publicly or privately.

**Reply by jeffmcneill** [reply]
"open core" not "open source"

**Comment by umar-mq** [comment]
> Good evening! While it is true that Microsoft owns GitHub, it is not accurate to say that it is a closed-source platform. GitHub's core product is built on top of Git, an open-source version control system, and it allows users to host their code repositories publicly or privately.

Git might be open-source, GitHub (the website), isn't.

**Comment by ToolmakerSteve** [comment]
The question that the title asks confuses two different things: The licensing of a repository (that anyone can put on github) versus the platform (github.com). One can post publicly available code with an "open source" license (MIT, etc.). Thus, one's repository is "open source".

It is irrelevant that github.com itself is not open source. If it ever went away, everyone would switch to some different alternative. Be glad a major company is willing to fund this!

**Reply by jeffmcneill** [reply]
Not irrelevant when wanting to migrate things that are part of the proprietary stack, such as wiki, issues, pipeline, etc. One cannot run their own Github instance (unless paying for enterprise). Compare this with Sourcehut and Codeberg / Forgejo which are fully open source.

**Reply by ksell8** [reply]
Microsoft was the first to sell software. And then they bought the server holding all the open source software and now is selling the open source regurgitated back to us.  evil empire.

**Comment by EasyCodeCoach** [comment]
Response #20633

**Comment by ay0ks** [comment]
This is just too naive, don't get it as an insult, but it is same thing as "Why bother using windows when it's closed source, use linux instead".

**Reply by jeffmcneill** [reply]
Your point is...?

**Reply by FlawLessError** [reply]
**ay0ks** probably means since it's the most famous why bother use the other services.
Nonetheless imo gitlab for personal projects and github for company projects.

**Comment by nazboyko** [comment]
When people say their project is "open source," they mean that the code is free for anyone to see, use, change, and share. This is about the rules (license) of the code itself.

GitHub, where many people put their open-source code, is a website that helps people work together on code. Even though GitHub is a closed-source platform (meaning its own code is not open for everyone to see or change), it is popular because it makes it easy to share and collaborate on open-source projects.

So, even though GitHub is closed source, people still use it for open-source projects because it offers good tools, is widely used, and has a big community. The "open source" label refers to the project's code, not to the platform it is hosted on.

**Reply by jeffmcneill** [reply]
Not accurate. GitHub itself has open source components (e.g., Git). It also hosts a variety of repositories which are not open source in the licensing sense. Yes, people use it for Open Source projects, but this conversation is about Github, not its hosted repositories.

**Reply by ay0ks** [reply]
> Not accurate. GitHub itself has open source components (e.g., Git). It also hosts a variety of repositories which are not open source in the licensing sense. Yes, people use it for Open Source projects, but this conversation is about Github, not its hosted repositories.

You're nit picking their words, GitHub does not need to be open-source for you to be able to host your projects there, also I'm not sure what you're trying to say with "It also hosts a variety of repositories which are not open source in the licensing sense.", that's the point of a repository hosting website, those projects are hosted here for their development teams to have better suite for developing those projects (not just plain Git), those teams may not be intending you to change or redistribute their code.

**Comment by rahil627** [comment]
hahahaha, this is actually legit issue. false advertising _(though, i'm certain they are careful in their wording..)._ nazboyko above with the only common sensical answer among a bunch of trolls ;)

but i _just_ tried gitlab (12/2025), and it's just a far inferior copy of github: similar marketing model (_not completely open-source, though has a free version that is.._), same push toward ai features for the sake of profit, but at the cost of the whole product's design being stuck with a PWA design from 10 years ago. microsoft is just constant proof that open-source projects need a strong leader (_though i too prefer anyone beyond the innovation-less corporation.. windows taskbar.. 🤦_).. and money to pay them.

however, i must say, codeburg is legit.. it keeps it simple! written in go, so you know it's extremely straight-forward, maintainable (_<3 ruby, but it requires a lot of care to use it on such a large project.._), as long as you trust those edm volunteer hipsters in berlin, germany with all your precious code/prose/writings.. one major caveat: 100mb max private repos, with a heavy FOSS bent.

**Reply by WalliGateshead** [reply]
Interesting perspective. Each platform has its place, but simplicity and sustainability matter more than hype. Open-source projects especially need strong direction and community trust to grow properly.
